src/core/webidl.js

```diff
@@ -114,27 +114,40 @@ function createIdlAnchor(escaped, data, parent) {
   const dfn = findDfn(data, name, {
     parent: parentName,
   });
+  const linkType = getDfnType(data.type);
   if (dfn) {
     if (!data.partial) {
       dfn.dataset.export = "";
-      dfn.dataset.dfnType = getDfnType(data.type);
+      dfn.dataset.dfnType = linkType;
     }
     return hyperHTML`<a
       data-link-for="${parentName.toLowerCase()}"
-      data-lt="${dfn.dataset.lt || null}">${escaped}</a>`;
+      data-link-type="${linkType}"
+      data-lt="${dfn.dataset.lt || null}"
+      >${escaped}</a>`;
   }
 
   const isDefaultJSON =
     data.type === "operation" &&
     data.name === "toJSON" &&
     data.extAttrs.some(({ name }) => name === "Default");
   if (isDefaultJSON) {
-    return hyperHTML`<a data-lt="default toJSON operation">${escaped}</a>`;
+    return hyperHTML`<a
+     data-link-type="dfn"
+     data-lt="default toJSON operation">${escaped}</a>`;
+  }
+  const unlinkedAnchor = hyperHTML`<a
+    data-idl="${data.partial ? "partial" : null}"
+    data-link-type="${linkType}"
+    data-title="${data.name}"
+    data-xref-type="${linkType}"
+    >${escaped}</a>`;
+
+  // Partial interfaces must always link somewhere else.
+  if (data.partial && !dfn) {
+    return unlinkedAnchor;
   }
 
-  const unlinkedAnchor = hyperHTML`<a data-xref-type="${
-    data.type
-  }">${escaped}</a>`;
   const showWarnings = name && data.type !== "typedef";
```

Comment:
```suggestion
   const showWarnings = name && data.type !== "typedef" && !(data.partial && !dfn);
```

I think this should work.